### PR TITLE
Added info on Dictionary and Lookup fields attributes

### PR DESCRIPTION
**Pull request by Vasyl-Martynov-WhiteDoc**
Added info on Optional, Allow custom values, Search, Sender attributes, added some screenshots and made some general improvements.

### CODE REVIEW

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst [hunk 1]
```diff
@@ -4,22 +4,64 @@ Dictionary field
...
+Dictionary is a list of records in a table form, e.g. list of employees (First name, Last name, Age, Address).
+Data from dictionary could be used in envelope.
 
-Dictionary - is a list of records in a systematic form, e.g. list of employee(Name, Surname, Age, Address).
-Data from dictionary could be used in envelope:
+Create Dictionary Field
+=======================
 
-1. Create dictionary field in template.
+1. Navigate to https://staging.whitedoc.space/template/form/editor
```
Comment (kolapse2) [inline docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst:13]: replace url with valid path (/template/form/create)

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst [hunk 2]
```diff
@@ -4,22 +4,64 @@ Dictionary field
...
-Dictionary envelope field
-=========================
-User need to select value from the list of records values. If allow custom value attribute is true - user can enter custom value.
-After value was set to field and processing flow was continued - value of dictionary field is blocked. Any updates in dictionary doesn't apply to this field.
\ No newline at end of file
+4. Fill the "Name" attribute of the field (should be unique per document).
+
+5. Select a role from "Role name" menu which will be filling this field.
+
+6. Fill the "Placeholder" attribute for this field (not mandatory, if you won't fill this field, name will be used as placeholder).
```
Comment (kolapse2) [inline docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst:27]: ..., if you won't fill this field, name will be used as placeholder). -> ..., if you don't fill this field, name will be used as placeholder

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst [hunk 3]
```diff
@@ -4,22 +4,64 @@ Dictionary field
...
+11. Choose input dictionary name in "Dictionary" field to select which dictionary will be used.
+
+12. Choose column from "Column name" list (it will appear after selecting a dictionary in previous step) of columns for this field to refer to.
+
+13. Click on "Save" will add new dictionary field.
+
+Edit Dictionary Field
+=====================
+
+1. Navigate to https://staging.whitedoc.space/template/form/editor
```
Comment (kolapse2) [inline docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst:46]: replace url with valid path (/template/form/create)

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst [hunk 4]
```diff
@@ -4,83 +4,94 @@ Lookup Field
...
 .. toctree::
 
-Lookup field automatically pull value from field dictionary field from record.
-Lookup field available only when template has one or more dictionaries without allow custom value attribute.
+Lookup field automatically pulls a value from related dictionary.
 
 Create Lookup Field
 ===================
 
 1. Navigate to https://staging.whitedoc.space/template/form/editor
```
Comment (kolapse2) [inline docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst:12]: replace url with valid path (/template/form/create)

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst [hunk 5]
```diff
@@ -4,83 +4,94 @@ Lookup Field
...
-4. Fill the name of the field (Should be unique per document)
+.. image:: pic_lookup/LookupCreationForm.png
+   :width: 600
+   :align: center
 
-5. Fill the placeholder for this field (Not mandatory, if you won't fill this field, name will be used as placeholder).
+4. Fill the "Name" attribute of the field (should be unique per document).
 
-6. Choose document from list of documents with dictionaries without allow custom value attribute.
+5. Fill the "Placeholder" attribute for this field (not mandatory, if you won't fill this field, name will be used as placeholder).
```
Comment (kolapse2) [inline docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst:24]: similar issue as on dictionary

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst [hunk 6]
```diff
@@ -4,83 +4,94 @@ Lookup Field
...
+10. Choose dictionary from "Related to" list of dictionaries for this field to refer to.
+
+11. Choose column from "Column name" list of columns for this field to refer to.
+
+12. Click on "Save" will add new lookup field.
 
 Edit Lookup Field
 =================
 
 1. Navigate to https://staging.whitedoc.space/template/form/editor
```
Comment (kolapse2) [inline docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst:43]: replace url with valid path (/template/form/create)

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst [hunk 7]
```diff
@@ -4,83 +4,94 @@ Lookup Field
...
+Lookup field can not be filled manually unless it has "Allow custom values" attribute enabled. To fill in lookup field value in related dictionary field should be selected. When value in dictionary is selected all related lookup fields will be automatically filled. If related dictionary is updated before user sends envelope an error message with information about dictionary changes will be shown. If dictionary was deleted all lookup fields which relate to this dictionary field will be highlighted with red.
```
Comment (kolapse2) [inline docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst:97]: If related dictionary has "Allow custom value" attribute and into dictionary field custom value added then lookup inherit this attribute and user will be able add custom data.

**Review by kolapse2 — CHANGES_REQUESTED** [review]
Please add changes that is required

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst (hunk 2 above)
Comment (Vasyl-Martynov-WhiteDoc) [reply docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst:27]: Done

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst (hunk 3 above)
Comment (Vasyl-Martynov-WhiteDoc) [reply docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/dictionary/dictionary.rst:46]: Done

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst (hunk 4 above)
Comment (Vasyl-Martynov-WhiteDoc) [reply docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst:12]: Done

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst (hunk 5 above)
Comment (Vasyl-Martynov-WhiteDoc) [reply docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst:24]: Done

File: docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst (hunk 7 above)
Comment (Vasyl-Martynov-WhiteDoc) [reply docs/source/pages/howToWorkWithPlatform/mainApplication/templateList/templateManagement/documentManagementOnTemplate/structuredDocument/fields/lookup/lookup.rst]: Done

**Pull request merged**